src/transformers/models/oneformer/image_processing_oneformer.py

```diff
@@ -990,6 +996,24 @@ def encode_inputs(
             encoded_inputs["class_labels"] = class_labels
             encoded_inputs["text_inputs"] = text_inputs
 
+        def get_input_ids(task_input: str) -> List[int]:
+            if task_input == "semantic":
+                # this task is semantic
+                return [49406, 589, 10549, 533, 29119, 1550, 49407]
+            elif task_input == "panoptic":
+                # this task is panoptic
+                return [49406, 589, 10549, 533, 1072, 24755, 49407]
+            elif task_input == "instance":
+                # this task is instance
+                return [49406, 589, 10549, 533, 34572, 49407]
+
+        task_input_ids = torch.zeros(len(task_inputs), 77, dtype=torch.int)
```

Comment:
We should make sure `77` is configurable here, as this can be modified in the model's config.

Reply:
This used to be hardcoded elsewhere, and I'm not sure where it came from, I'll double check.

Reply:
Ok, I looked again. 

The truth is that this is going to be peculiar no matter what given the current state of things. Correct me if I'm wrong:

the `task_inputs` needed by the OneFormer, is some kind of `ids` representing some text to represent the given task (semantic,panoptic,instance).

- First weirdness. This model expects having sentences be exactly `this task is semantic` and encoded into ids (not embedded yet). This is usually done by the tokenizer, but what is this string supposed to be ? There is no way to get it in the conf, and it seems pretty hardcoded https://github.com/huggingface/transformers/blob/main/src/transformers/models/oneformer/processing_oneformer.py#L128.
- Since the original strings are hardcoded, I don't see a reason to actually use the tokenizer here. We could do it, see below. 
- pipelines **will** not ever use the Processor. Processor just don't have any form of unified call signature, making using them extremely bug prone.
- Since the tokenizer would be used only to tokenizer 3 extremely specific strings, with an extremely specific tokenizer with that specific 77 length. I thought that hardcoding everything in the `ImageProcessor` was just the most logical thing to do.

Is there any actual other usage of the tokenizer beyond those 3 strings? I'm really confused about this.

If we want the cleaner thing, we need a way for either the model, or the tokenizer or the image processor to know about those 3 specific strings. And I will figure out a way to make the pipeline discover that a tokenizer is needed and needs to be appended to the model_inputs.

Reply:
Thanks for bringing it up, it forced me to look more closely and I realized I broke the preprocessing during the rebase. It's now fixed.

Reply:
OK. So digging around the processor and image processor - I think I understand a bit more about what's happening. I've replied as best I can to your points above. 

> the task_inputs needed by the OneFormer, is some kind of ids representing some text to represent the given task (semantic,panoptic,instance).

Yes, this is my understanding too.

> First weirdness. This model expects having sentences be exactly this task is semantic and encoded into ids (not embedded yet). This is usually done by the tokenizer, but what is this string supposed to be ? There is no way to get it in the conf, and it seems pretty hardcoded https://github.com/huggingface/transformers/blob/main/src/transformers/models/oneformer/processing_oneformer.py#L128.

Yes, I think there's two hardcoded strings: 
* `"the task is {TASK}"`
* `"a photo of a {CLS}"` e.g. [here](https://github.com/huggingface/transformers/blob/f424b09410863eb9bb002bbe800fe86758fba839/src/transformers/models/oneformer/image_processing_oneformer.py#L802) 

> Since the tokenizer would be used only to tokenizer 3 extremely specific strings, with an extremely specific tokenizer with that specific 77 length. I thought that hardcoding everything in the ImageProcessor was just the most logical thing to do.
> Is there any actual other usage of the tokenizer beyond those 3 strings? I'm really confused about this.

Yes, unfortunately I think so as the `"a photo of a {CLS}"`  strings are also tokenized in the [processor here](https://github.com/huggingface/transformers/blob/14d058b9404b2a0659441038e120508e4a9ae10c/src/transformers/models/oneformer/processing_oneformer.py#L139). We need the tokenizer in case the user changes the classes the model is trained on. The classes are determined in `self.metadata` (e.g. [here](https://github.com/huggingface/transformers/blob/f424b09410863eb9bb002bbe800fe86758fba839/src/transformers/models/oneformer/image_processing_oneformer.py#L829) so can change). 


> pipelines will not ever use the Processor. Processor just don't have any form of unified call signature, making using them extremely bug prone.

This is tough. I don't think we should add e.g. text processing into the image processor because the API for the processor is hard to use in pipelines: the image processor shouldn't know anything about pipelines. It might be difficult because of the nature of processors but would be good to know more about these pain points. Are there changes that could happen where this would be possible? Probably for a conversation outside this PR.

Reply:
`"a photo of a {CLS}"` afaik, this is not used by the pipeline ever. My guess is that this is only used at training time (couldn't really figure it out when this was useful tbh).

> text processing into the image processor because the API for the processor is hard to use in pipelines

100% !. I don't claim I'm doing it because it's hard, I'm claiming it's because there's no real text processing going on (since everything is fixed, and there's only 3 variants).

>  Are there changes that could happen where this would be possible? 

I don't think so. `Processor` doesn't have a unified call signature, and I don't think it should. In my mind they exist to make nice shareable snippets to use a particular model. And for that they work great. But it doesn't really work to build upon in a library, because there's no way to know for an arbitrary `Processor` how to call it. Am i supposed to send images, texts, both , `task_inputs`, etc...

Happy to discuss further.

I'll do the "cleaner" approach which uses the tokenizer. You'll see that the issue will be that there would be the need for a special method on the tokenizer to include the `task_inputs`. Something like: https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/conversational.py#L257
I don't have any issue adding such specific method, but it's unlikely to be used by anything else than `oneformer`.

In practice, since oneformer is using `CLIPTokenizer`. This special method has to spill in `CLIPTokenizer` and it felt worse to spill this into an unrelated class than hardcoding the IDs (which do avoid a ton of computation).

Since this is causing debate, I'll change strategy and remove `Oneformer` from this PR (which main thing is actually just doing `ImageProcessor` modification.) 

And leave the `OneFormer` support for another PR.